Code/src/main/java/nl/utwente/group10/ui/components/Block.java

```diff
@@ -1,16 +1,23 @@
 package nl.utwente.group10.ui.components;
 
+import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.gestures.GestureCallBack;
+import nl.utwente.group10.ui.gestures.UIEvent;
+import javafx.event.EventType;
+import javafx.scene.Node;
 import javafx.scene.layout.StackPane;
 
 /**
  * Base UI Component that other visual elements will extend from.
  * If common functionality is found it should be refactored to here.
  */
-public class Block extends StackPane {
+public class Block extends StackPane implements GestureCallBack {
 
 	/** Selected state of this Block*/
 	private boolean isSelected = false;
 	
+	protected static CustomUIPane cup;
+	
```

Comment:
Question:
What is this for? I cannot tell why the Block needs to store the CustomUIPane.
Request:
Add javaDoc, it could clarify the usage of some code.